mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -64,6 +67,18 @@ public CassandraAnnotationMapper(CassandraId mailboxId, Session session) {
             .collect(Collectors.toList());
     }
 
+    public List<MailboxAnnotation> getAnnotationsByKeysWithOneDepth(Set<String> keys) {
```

Comment:
Like this, you do up to n sequential reads to cassandra.
IMO, you should rely on futures for getAnnotationsByKeyWithOneDepth